vim_configurable: change python version detection

KaiSforza opened the pull request:
Instead of explicitly stating "I want version 3" or "I want version 2"
you now simply specify what the `python` argument will be, and
vim_configurable will set up the flags for you. `config.vim.python` must
be set, still.

This is a complete rewrite of the PR I submitted in June 2015.

(Branch name is a bit of a misnomer, but that's the name of the package I use to test.)

KaiSforza commented:
A better solution than #7619

jagajaga commented:
cc me

KaiSforza commented:
Any input on this?

jagajaga commented on pkgs/applications/editors/vim/configurable.nix:
```diff
@@ -62,15 +64,16 @@ composableDerivation {
     nativeBuildInputs
-      = [ ncurses pkgconfig gtk libX11 libXext libSM libXpm libXt libXaw libXau
-          libXmu glib libICE ];
+      = if args.gui != "no" then [ ncurses pkgconfig gtk libX11 libXext libSM libXpm libXt libXaw libXau
```
You can take out `ncurses pkgconfig`.

KaiSforza replied:
From both?

jagajaga commented on pkgs/applications/editors/vim/configurable.nix:
```diff
@@ -92,26 +95,9 @@ composableDerivation {
+        feat = "python${pythonVersion}interp";
         enable = {
-          nativeBuildInputs = [ python ];
-        } // lib.optionalAttrs stdenv.isDarwin {
```
Why you remove darwin part?

KaiSforza replied:
It's not needed, at least in my experience. It automatically checks for the location, though I actually didn't test with python2...

KaiSforza replied:
Just tested, it goes to the /nix directory for this stuff.

jagajaga replied:
What do you mean?

KaiSforza replied:
This just removes the `--with-python-config-dir` configuration flag, which is not needed, as it pulls in the correct location during the configuration process.

KaiSforza committed:
vim-configurable: Change python versioning

Instead of explicitly stating "I want version 3" or "I want version 2"
you now simply specify what the python argument will be, and
vim_configurable will set up the flags for you. config.vim.python must
be set, still.

KaiSforza commented:
Complete rewrite, much simpler, no changes to X.

The python3 specific stuff (python3Support, etc) was removed entirely, in favor of a simple if/else statement using `python ? isPy3`.

KaiSforza commented:
Pinging @jagajaga.

jagajaga commented:
Hmm. What does this statement do?

KaiSforza commented:
places a 3 if `python ? isPy3` returns true, otherwise leaves it blank. It then does the opposite for the `--disable` flag (at least in darwin).

KaiSforza commented:
The `? isPy3` means that it will return false if that attribute is not set:
```
nix-repl> python ? isPy3
false

nix-repl> python3 ? isPy3
true
```

jagajaga commented:
So to enable python 3 we need to override pkg with `python = python3`?

KaiSforza commented:
Yes. Changing the `python` input will change what the package enables or disables.
